shared/domain/src/test/java/com/ivy/domain/usecase/balance/BalanceBuilderTest.kt

```diff
@@ -0,0 +1,300 @@
+            expected = mapOf(
+                EUR to nonZeroDouble(10.0)
+            )
+        ),
+        TwoInDiffCurrencyDepositsFromIncomeAndTransfersIn(
+            values = AccountStats(
+                income = StatSummary(
+                    trnCount = count(2),
+                    values = mapOf(
+                        EUR to positiveDouble(3.14)
+                    )
+                ),
+                transfersIn = StatSummary(
+                    trnCount = count(1),
+                    values = mapOf(USD to positiveDouble(50.0))
+                ),
+                expense = StatSummary(
+                    trnCount = NonNegativeInt.Zero,
+                    values = StatSummary.Zero.values
+                ),
+                transfersOut = StatSummary(
+                    trnCount = NonNegativeInt.Zero,
+                    values = StatSummary.Zero.values
+                ),
+            ),
+            expected = mapOf(
+                EUR to nonZeroDouble(10.0),
+                USD to nonZeroDouble(50.0)
+            )
+        ),
+        TwoDepositsInDiffCurrencyIncomeAndOneTransferIn(
+            values = AccountStats(
+                income = StatSummary(
+                    trnCount = count(2),
+                    values = mapOf(
+                        USD to positiveDouble(50.0),
+                        EUR to positiveDouble(3.14)
+                    )
+                ),
+                transfersIn = StatSummary(
+                    trnCount = count(1),
+                    values = mapOf(
+                        GBP to positiveDouble(0.5)
+                    )
+                ),
+                expense = StatSummary(
+                    trnCount = NonNegativeInt.Zero,
+                    values = StatSummary.Zero.values
+                ),
+                transfersOut = StatSummary(
+                    trnCount = NonNegativeInt.Zero,
+                    values = StatSummary.Zero.values
+                ),
+            ),
+            expected = mapOf(
+                EUR to nonZeroDouble(3.14),
+                USD to nonZeroDouble(50.0),
+                GBP to nonZeroDouble(0.5)
+            )
+        )
+    }
+
+    @Test
+    fun `when stats are empty`() {
+        // given
+        val testCase = ValuesTestCase.Empty
+        val balanceBuilder = BalanceBuilder()
+
+        // when
+        val stats = testCase.values
+
+        balanceBuilder.processDeposits(
+            incomes = stats.income.values,
+            transfersIn = stats.transfersIn.values
+        )
+        balanceBuilder.processWithdrawals(
+            expenses = stats.expense.values,
+            transfersOut = stats.transfersOut.values
+        )
+        val balance = balanceBuilder.build()
+
+        // then
+        balance shouldBe testCase.expected
+    }
+
+    @Test
+    fun `process one deposit come from incomes`() {
+        // given
+        val testCase = ValuesTestCase.OneDepositFromIncome
```

Comment:
Use `@TestParameter` and `@RunWith(TestParameterInjector::class)` search it in my other tests

Reply:
There should be one test with multiple test cases